### PR TITLE
TestPuzzlePieceCreation: Add unit tests for rotate function

### PR DESCRIPTION
**Pull request by cnchengucsd**

### CODE REVIEW

File: tests/test_puzzle_piece_creation.py [hunk 1]
```diff
@@ -1,17 +1,62 @@
...
+    angle = math.radians(180)
+    expected_point = (-1, -1)
+    assert rotate(origin, point, angle) == pytest.approx(expected_point)
+
+
+def test_rotate_neg_45_degrees():
+    """Test case 3: Rotate -45 degrees."""
+    origin = (0, 0)
+    point = (1, 1)
+    angle = math.radians(-45)
```
Comment (Bozhie) [inline tests/test_puzzle_piece_creation.py:57]: nice

**Review by msimbirsky — APPROVED** [review]
LGTM

File: tests/test_puzzle_piece_creation.py [hunk 2]
```diff
@@ -1,17 +1,62 @@
...
+    expected_x = math.sqrt(2)
+    expected_y = 0
```
Comment (Bozhie) [inline tests/test_puzzle_piece_creation.py:59]: it approaches the y axis -- maybe add a comment to explain this delta (starts at 45 degree angle then ends up on the x-axis)

**Review by Bozhie — APPROVED** [review]
this is nice

**Pull request merged**